spec/controllers/spree/api/products_controller_spec.rb

```diff
@@ -176,5 +168,66 @@ module Spree
         end
       end
     end
+
+    describe '#bulk_products' do
+      context "as an enterprise user" do
+        let!(:taxon) { create(:taxon) }
+        let!(:product2) { create(:product, supplier: supplier, primary_taxon: taxon) }
+        let!(:product3) { create(:product, supplier: supplier2, primary_taxon: taxon) }
+        let(:current_api_user) { supplier_enterprise_user(supplier) }
+
+        before { current_api_user.enterprise_roles.create(enterprise: supplier2) }
+
+        it "returns a list of products" do
+          spree_get :bulk_products, { page: 1, per_page: 15 }, format: :json
+
+          expect(returned_product_ids).to eq [product3.id, product2.id, product1.id]
+        end
+
+        it "returns pagination data" do
+          spree_get :bulk_products, { page: 1, per_page: 15 }, format: :json
+
+          expect(json_response['pagination']).to eq "results" => 3, "pages" => 1, "page" => 1, "per_page" => 15
+        end
+
+        it "filters results by supplier" do
+          spree_get :bulk_products, { page: 1, per_page: 15, q: {supplier_id_eq: supplier.id} }, format: :json
+
+          expect(returned_product_ids).to eq [product2.id, product1.id]
+        end
+
+        it "filters results by product category" do
+          spree_get :bulk_products, { page: 1, per_page: 15, q: {primary_taxon_id_eq: taxon.id} }, format: :json
+
+          expect(returned_product_ids).to eq [product3.id, product2.id]
+        end
```

Comment:
we should test pagination a bit better I think. maybe a test where page is not 1. and then test some limits like request page over the limit (page 2 of 50 records where per_page is 50).

Reply:
I've used a smaller number of objects (2 per page), but added page tests. :+1: